File: src/tools/get-contacts.ts
/**
 * Herramienta MCP para obtener contactos de Track HS
 */

import { BaseTrackHSTool } from '../core/base-tool.js';
import { ContactsResponse, GetContactsParams } from '../types/contacts.js';

export class GetContactsTool extends BaseTrackHSTool {
  name = 'get_contacts';
  description = 'Retrieve all contacts from Track HS CRM system. Contacts include guests, owners, or vendor employees.';
  
  inputSchema = {
    type: 'object' as const,
    properties: {
      sortColumn: { 
        type: 'string', 
        enum: ['id', 'name', 'email', 'cellPhone', 'homePhone', 'otherPhone', 'vip'],
        description: 'Sort by id, name, email, mobile phone, home phone, other phone, vip'
      },
      sortDirection: { 
        type: 'string', 
        enum: ['asc', 'desc'], 
        default: 'asc',
        description: 'Sort ascending or descending'
      },
      search: { 
        type: 'string', 
        description: 'Search by first name, last name, email, mobile phone, home phone, other phone with right side wildcard'
      },
      term: { 
        type: 'string', 
        description: 'Locate contact based on a precise value such as ID or name'
      },
      email: { 
        type: 'string', 
        format: 'email',
        description: 'Search contact by primary or secondary email address'
      },
      page: { 
        type: 'number', 
        description: 'Page Number',
        minimum: 1
      },
      size: { 
        type: 'number', 
        description: 'Page Size',
        minimum: 1,
        maximum: 100
      },
      updatedSince: { 
        type: 'string', 
        format: 'date-time', 
        description: 'Date in ISO 8601 format. Will return all contacts updated since timestamp'
      }
    },
    required: []
  };

  async execute(params: GetContactsParams = {}): Promise<ContactsResponse> {
    // Validar parámetros
    this.validateParams(params);

    // Construir query parameters
    const queryParams = new URLSearchParams();
    
    // Aplicar valores por defecto
    const sortDirection = params.sortDirection || 'asc';

    // Agregar parámetros a la query
    if (params.sortColumn) {
      queryParams.append('sortColumn', params.sortColumn);
    }
    queryParams.append('sortDirection', sortDirection);

    if (params.search) {
      queryParams.append('search', params.search);
    }

    if (params.term) {
      queryParams.append('term', params.term);
    }

    if (params.email) {
      queryParams.append('email', params.email);
    }

    if (params.page) {
      queryParams.append('page', params.page.toString());
    }

    if (params.size) {
      queryParams.append('size', params.size.toString());
    }

    if (params.updatedSince) {
      queryParams.append('updatedSince', params.updatedSince);
    }

    const endpoint = `/crm/contacts?${queryParams.toString()}`;
    
    try {
      const result = await this.apiClient.get<ContactsResponse>(endpoint);
      return result;
    } catch (error) {
      throw new Error(`Error al obtener contactos: ${error instanceof Error ? error.message : 'Error desconocido'}`);
    }
  }
}

File: src/trackhs_mcp/tools/get_contacts.py
"""
Herramienta MCP para obtener contactos de Track HS API
"""

from typing import Optional
from ..core.api_client import TrackHSApiClient

def register_get_contacts(mcp, api_client: TrackHSApiClient):
    """Registra la herramienta get_contacts"""
    
    @mcp.tool()
    async def get_contacts(
        page: int = 1,
        size: int = 10,
        sort_column: str = "id",
        sort_direction: str = "asc",
        search: str = None,
        term: str = None,
        email: str = None,
        updated_since: str = None
    ):
        """
        Get contacts from Track HS
        
        Args:
            page: Número de página (default: 1)
            size: Tamaño de página (default: 10)
            sort_column: Columna para ordenar (default: "id")
            sort_direction: Dirección de ordenamiento (default: "asc")
            search: Término de búsqueda (opcional)
            term: Término específico (opcional)
            email: Email del contacto (opcional)
            updated_since: Filtro por fecha de actualización (opcional)
        """
        query_params = {
            "page": page,
            "size": size,
            "sortColumn": sort_column,
            "sortDirection": sort_direction
        }
        
        if search:
            query_params["search"] = search
        if term:
            query_params["term"] = term
        if email:
            query_params["email"] = email
        if updated_since:
            query_params["updatedSince"] = updated_since
        
        endpoint = f"/crm/contacts"
        query_string = "&".join([f"{k}={v}" for k, v in query_params.items()])
        if query_string:
            endpoint += f"?{query_string}"
        
        try:
            result = await api_client.get(endpoint)
            return result
        except Exception as e:
            return {"error": f"Error al obtener contactos: {str(e)}"}
